lib/rules/max-line-length/index.js

```diff
@@ -59,14 +59,14 @@ const rule = (primary, secondaryOptions, context) => {
 		let skippedSubStrings = [];
 		let skippedSubStringsIndex = 0;
 
-		for (const pattern of EXCLUDED_PATTERNS)
-			for (const match of execall(pattern, rootString)) {
-				const subMatch = match.subMatches[0] || '';
-				const startOfSubString = match.index + match.match.indexOf(subMatch);
+		for (const pattern of EXCLUDED_PATTERNS) {
+			for (const match of rootString.matchAll(pattern)) {
```

Comment:
[note] [The MDN doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll#examples) says:

> `matchAll` internally makes a clone of the `regexp`

so we don't need to worry about mutable `RegExp` instances (`lastIndex` never changes).